### DISCUSSION TITLE
Immediately enter app on SSO login

### DISCUSSION THREAD

**Discussion by TheAzack9** [directus/directus#5471] [Feature Requests]
Hello again :)

1) The issue and what you expected to happen
    When logging in with SSO do you get redirected back to the login page, which in turn redirects to the "continue as" screen. This is a more jarring experience than getting redirected to the last page as when you log in with email + password. I feel having it work the same with SSO would be a better experience and should be a relatively small chamge. :)

Problem is somehow changing https://github.com/directus/directus/blob/main/app/src/routes/login/components/sso-links.vue#L37 to pass something other than window.location.href. One might need a separate parameter since we still don't know the email address of the user until after the SSO callback. 

The part of code redirecting is here: https://github.com/directus/directus/blob/76b31eda78d53f7866040f897376fa514e7b1202/api/src/controllers/auth.ts#L265-L281

2) Exact steps to reproduce this issue
    Setup some kind of SSO provider and the issue should be replicated pretty easily.

3) Your environment:
Directus version: 9.0.0 rc 45 (although since i went through the source am i pretty sure the issue still exists in rc51 :) )
DBMS: MySQL 8
Deployment: Docker
Browser: Chrome 89

**Comment by benhaynes** [comment]
I agree, it would be a better user experience to log them automatically/immediately once authenticated. I'm down for this, as long as it doesn't add a significant complexity to the SSO configuration. 💯

@TheAzack9 , any interest in trying a PR?

**Reply by rijkvanzanten** [reply]
The main technical issue here is that the SSO flow only sets a refresh token (by design), so the app has to "refresh" on first load in order to fetch the access token and current user. Right now, the first load is nothing different from a regular browser refrehs

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
